### ISSUE TITLE
Integrate othree html5 plugin for the HTML highlighting? 

### ISSUE THREAD

**Issue by dman777** [jonsmithers/vim-html-template-literals#3]
Thanks for the effort in this. 

The highlighting is a bit off... for instance, a opening `<div>` tag would be a different color than the closing `</div>` tag. 

Looking at the source code it is pulling all of it coloring from vim-javascript(which is great for Js...it is my favorite plugin). 

Is there a way to also integrate https://github.com/othree/html5.vim for the HTML highlighting portion? The html5 plugin is much cleaner and does a better job on the HTML portion than the JS does. 

I spent about 5 hours trying to make it work with no success.

**Comment by dman777** [comment]
UPDATE: I found a way to incorporate the syntax objects from html5.vim to  vim-html-template-literals. I was going to submit a PR with it but still needs some work... not sure how much work is worth putting into if the indention doesn't work. 

@jonsmithers  is there anyway you can get the indention working? I took a stab at it  and didn't get anywhere.

**Comment by jonsmithers** [comment]
I think it's normal for `</div>` to be a different color than `<div>`. A lot of colorschemes seem to do that intentionally for xml syntax.

It would be nice to use html syntax instead of xml syntax. When I initially tried using html, I would get an infinite loop issue because javascript syntax would include html syntax which would then re-include javascript syntax. Maybe you've seen this too? I think this can be avoided by including html syntax inside a filetype autocommand instead of in the javascript syntax definition. This is how [cdata/vim-tagged-template](https://github.com/cdata/vim-tagged-template) applies custom js syntax and thus avoids the infinite loop issue.

All I'm really saying is that we could try using html syntax by writing the js syntax in a filetype autocommand instead of inside `syntax/javascript.vim`. Any html syntax plugins like [othree/html5](othree/html5.vim) would then apply naturally.

```vim
autocmd FileType javascript,typescript : call htmltemplateliterals#amendSyntax()
```

I've had to write custom code to make css work inside `<style>` tags. I could delete that code if we switch from xml to actual html because it will just work.

**Comment by jonsmithers** [comment]
Regarding indentation....that's been a real challenge. Most of the issues arise at the seems between 2 different syntaxes (``html`...` `` or `${...}` or `<style>...</style>`), and I just haven't been able to find a simple algorithm that handles everything correctly. There's a lot of edge cases. I'm definitely open to ideas. Or re-writing `indent/javascript.vim` from scratch.

Switching from xml to html would help by taking care of all the `<style>` tag indentation.

**Comment by dman777** [comment]
> I think it's normal for </div> to be a different color than <div>. A lot of colorschemes seem to do that intentionally for xml syntax.

I dont know about XML, but not for HTML... pygmentize does not and nether does html5.vim. But I have this fixed already.

> Switching from xml to html would help by taking care of all the <style> tag indentation.

I switched from xml to html for syntax. Good news! I got indention working for vhti(this plugin). I gutted out the function `ComputeLitHtmlIndent`. Then inside of it I cut pasted all the indent functions  from html.vim. Indention works great! At least from initial testing. 

Right now I am working on improving the syntax highlighting and add syntax objects from  `vim-javascript` to it so they can be themed/colorized better.

This is a blocker for me to upgrade my app to Polymer 3, so I will be working on this a little each day after work. I should have something solid in about a week and I can submit a PR, does that sound good?

**Comment by jonsmithers** [comment]
Differently start/end colors is normal for xml, but not so much for html.

It'd be awesome to have some more hands in this code! Would you mind looking at the approach I took in PR #4? I think you can simply invoke `HtmlIndent()` from `indent/html.vim` rather than copying and pasting code into `ComputeLitHtmlIndent()`.

**Comment by dman777** [comment]
Sure, I am still very new to VIM coding syntax though. Can you go ahead and merge that PR and I will work off of it?

**Comment by dman777** [comment]
@jonsmithers  Your PR really improved the syntax! Mass improvement! With your last PR I would say that it puts this plugin in a full working state ready to be used without any deficiencies. 

I submitted the PR with the indention from html.vim. I still found some inconsistencies with the indention, but using the html.vim code from the PR fixes them. Feel free to clean it up...I left some notes in the PR about why I did it the way it is.
